### DISCUSSION TITLE
How to deploy two testates on a server to correspond to two vehicles with different accounts

### DISCUSSION THREAD

**Discussion by lrx1021** [teslamate-org/teslamate#3460] [General]
I have two cars, each with two different accounts. I would like to ask for advice on how to use two sets of testamate on a Linux server at the same time. Currently, I have already installed one. Sincerely ask for a detailed instruction，thanks

**Comment by brianmay** [comment]
I think if you have two accounts, you will, currently, need to instances of Teslamate.

Not sure if it is OK to share the same database or not, might be risky. Would need to check the database schema to be sure. Might be OK if cars are referenced by the unique id assigned by Tesla. (and I think we might do this).

I think you can use the same MQTT broker, but you would need to set the MQTT_NAMESPACE env variable to each instance to have a difference value (wondering if we should use the car id not the car number in the MQTT topic, but I think that is a discussion for another time).

**Comment by DrMichael** [comment]
Wouldn't it be easiest to run two containers. You then just would have to adjust the ports, which are exposed to the host...

**Reply by lrx1021** [reply]
I also think so, but I'm not very familiar with Docker. I don't know how to run two containers independently, ahhh. I'm not sure if I could take the liberty of asking for detailed teaching from you

**Comment by cwanja** [comment]
Can definitely have multiple vehicles on a single instance. 

If you do not want to have two TeslaMate instances, you would need to add one account to the other as a secondary driver. And then refresh TeslaMate and ensure the account that has both cars is logged into TeslaMate.

**Reply by lrx1021** [reply]
I think the two examples are the solution I want, but I don't know how to do it

**Reply by cwanja** [reply]
Which is your desired solution?

**Reply by lrx1021** [reply]
Currently, a set of testamate has been installed and a model Y has been monitored. I would like to add an instance on this Linux server to independently monitor my other model 3, which are different Tesla accounts

**Reply by cwanja** [reply]
Create a new folder, create a new yml file, modify the ports (4001 instead of 4000) and start it up like your other instance. Login with the other account.

Not sure if the database will have port issues or just seamlessly run. Never tried it. But running multiple docker instances is not unheard of.

**Reply by a350844866** [reply]
i have the same question,i have three tesla in my family((three tesla account),i use docker-compose to start。should i copy docker-compose to three files,and run three times .or i can add three tesla instance in one docker-compose file,and use the same grafana and postgres?

**Comment by rhummer** [comment]
If the cars are in the same house hold, why not just make another Tesla account and make that account an additional driver for the vehicle(s)?  So all the vehicles are present under one account. I've done this for my household. Our vehicles are under one account anyway. But when I have to deal with Tesla Support/Service and need to disable TeslaMate (ie: Tesla says change the password to force all tokens to expire) I much prefer resetting the password on that account than my primary account.

**Reply by brianmay** [reply]
Probably that would be the idea solution. But I think there are some good reasons why you might not want to move all vehicles to one account (not applicable for me personally, I only have one car).
